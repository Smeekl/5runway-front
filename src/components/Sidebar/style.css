.sidebar {
  box-sizing: content-box;
  background-color: var(--color-secondary-white);
  z-index: 1;
  top: 0;
  right: 0;
  animation-duration: 0.5s;
  animation-name: slideOpen;
  animation-fill-mode: forwards;
}

@keyframes slideOpen {
  0% {
    width: 0%;
  }
  100% {
    height: 100%;
    width: 413px;
  }
}

.close {
  animation-duration: 0.3s;
  animation-fill-mode: forwards;
  width: 413px;
  animation-name: slideClose;
}

@keyframes slideClose {
  0% {
    width: 413px;
  }
  100% {
    width: 0px;
  }
}

.sidebar__header {
  padding: 26px 35px 0 35px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.sidebar__wrapper {
  background-color: var(--color-secondary-white);
  border-right: 5px solid var(--color-primary);
  display: flex;
  z-index: 99;
  width: 413px;
  height: 100%;
  flex-direction: row-reverse;
  justify-content: flex-start;
}
.sidebar__wrapper_main {
  display: flex;
  z-index: 2000;
  width: 413px;
  height: 100%;
  flex-direction: row-reverse;
  justify-content: flex-start;
}
