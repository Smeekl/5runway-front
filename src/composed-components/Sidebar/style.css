.bars {
  background-color: var(--color-primary-transparent);
  z-index: 2;
  width: 100%;
  height: 100%;
  position: fixed;
  display: flex;
  flex-direction: row-reverse;
  top: 0;
  right: 0;
}
