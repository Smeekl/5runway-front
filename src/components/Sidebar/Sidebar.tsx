import { CloseOutlined, RightOutlined } from '@ant-design/icons';
import React, { useState } from 'react';
import cn from 'classnames';
import { useLocation, useNavigate } from 'react-router-dom';
import { ReactComponent as IconSidebarLogo } from '@svgIcons/iconSidebarLogo.svg';
import { useAuth } from '@providers/auth.provider';
import './style.css';

interface ISidebarProps {
  children: React.ReactElement;
  title: string;
  index: number;
}

const Sidebar = ({ children, title, index }: ISidebarProps) => {
  const [isClosing] = useState(false);
  const { isLoggedIn } = useAuth();
  const navigate = useNavigate();
  const { pathname } = useLocation();

  const clickHandlerCloseLast = () => {
    const currPathArray = pathname.split('/');
    currPathArray.pop();

    const newPath = currPathArray.join('/') ? currPathArray.join('/') : '/';
    navigate(newPath, { replace: true });
  };

  const clickHandlerCloseAll = () => {
    const newPath = '/';
    navigate(newPath, { replace: true });
  };

  return (
    <div
      className={cn({
        close: isClosing,
        sidebar__wrapper_main: !isClosing && index === 0,
        sidebar__wrapper: !isClosing && index !== 0,
      })}
    >
      <div
        className="sidebar"
        onClick={(e) => {
          e.stopPropagation();
        }}
      >
        <div className="sidebar__header">
          {isLoggedIn ? <IconSidebarLogo /> : <span>{title}</span>}
          {index === 0 ? (
            <span>
              <CloseOutlined onClick={clickHandlerCloseAll} />
            </span>
          ) : (
            <span>
              <RightOutlined onClick={clickHandlerCloseLast} />
            </span>
          )}
        </div>
        {children}
      </div>
    </div>
  );
};

export default Sidebar;
