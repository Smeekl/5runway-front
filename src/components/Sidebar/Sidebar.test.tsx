import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Sidebar from './Sidebar';

describe('Sidebar  UI tests', () => {
  test('should render corectly with props', () => {
    const { container } = render(
      <MemoryRouter>
        <Sidebar children={<div>example</div>} title={'example'} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should render corectly without props', () => {
    const { container } = render(
      <MemoryRouter>
        <Sidebar />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const { container } = render(
      <MemoryRouter>
        <Sidebar />
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });

  test('should close on click close button', async () => {
    const history = createMemoryHistory({
      initialEntries: ['/button'],
    });
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Sidebar children={<div>button</div>} title={'example'} index={0} />
      </Router>
    );

    await waitFor(() => {
      userEvent.click(getByTestId('close-button'));
    });
    expect(history.location.pathname).toBe('/');
  });
});
