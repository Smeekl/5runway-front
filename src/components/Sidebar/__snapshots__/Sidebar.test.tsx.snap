// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar  UI tests should render corectly with props 1`] = `
<div>
  <div
    class="sidebar__wrapper"
  >
    <div
      class="sidebar"
    >
      <div
        class="sidebar__header"
      >
        <span>
          example
        </span>
        <span>
          <span
            aria-label="right"
            class="anticon anticon-right"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div>
        example
      </div>
    </div>
  </div>
</div>
`;

exports[`Sidebar  UI tests should render corectly without props 1`] = `
<div>
  <div
    class="sidebar__wrapper"
  >
    <div
      class="sidebar"
    >
      <div
        class="sidebar__header"
      >
        <span />
        <span>
          <span
            aria-label="right"
            class="anticon anticon-right"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
